Add a Security Group for Windows AMI Creation

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR adds a security group to use for the manual portion of Windows AMI creation. It allows RDP access and has a name to make it easy to reference when spinning up an instance with a VM import AMI.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Access is needed to finish the setup for Windows AMIs after the initial import from VM. This provides a static security group to use in a convenient way since this process is currently manual.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
Automated tests pass. I have applied these changes to the Staging environment and was able to use this Security Group to access an instance for configuration.
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

mcdonnnj committed:
Add a security group for Windows AMI creation

Add a security group that will allow RDP access to instances so they can be
accessed to finalize configuration for Windows AMI creation.

jsf9k commented on images/windows_ami_sg.tf:
```diff
@@ -0,0 +1,32 @@
+  }
+}
+
+# Allow ingress to instances via RDP (TCP)
+resource "aws_security_group_rule" "windows_ami_ingress_via_rdp_over_tcp" {
+  security_group_id = aws_security_group.windows_ami.id
+  type              = "ingress"
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  from_port         = 3389
+  to_port           = 3389
+}
+
+# Allow ingress to instances via RDP (UDP)
+resource "aws_security_group_rule" "windows_ami_ingress_via_rdp_over_udp" {
+  security_group_id = aws_security_group.windows_ami.id
+  type              = "ingress"
+  protocol          = "udp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  from_port         = 3389
+  to_port           = 3389
+}
```
```suggestion
# Allow ingress to instances via RDP
resource "aws_security_group_rule" "windows_ami_ingress_via_rdp" {
  for_each = toset(["tcp", "udp"])

  security_group_id = aws_security_group.windows_ami.id
  type              = "ingress"
  protocol          = each.value
  cidr_blocks       = ["0.0.0.0/0"]
  from_port         = 3389
  to_port           = 3389
}
```

jsf9k reviewed (changes requested):
I made one suggestion to avoid duplicating some code.

mcdonnnj committed:
DRY out some Security Group rules

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

dav3r commented on images/variables.tf:
```diff
@@ -141,3 +141,9 @@ variable "third_party_bucket_name_prefix" {
+
+variable "windows_ami_sg_name" {
+  type        = string
+  description = "The name to associate with the Security Group that allows access for finalizing Windows AMI configuration."
```
I don't think we capitalize "security group" - AWS doesn't do that in their docs:
```suggestion
  description = "The name to associate with the security group that allows access for finalizing Windows AMI configuration."
```

mcdonnnj replied:
Please see 97ee8a1 (I did it manually to update the README at the same time).

dav3r commented on images/windows_ami_sg.tf:
```diff
@@ -0,0 +1,24 @@
+# Security group for manually finishing AMI setup for imported Windows AMIs
+resource "aws_security_group" "windows_ami" {
+  # We include a name for convenience because this Security Group will be
```
Unnecessary capitalization:
```suggestion
  # We include a name for convenience because this security group will be
```

mcdonnnj replied:
Please see 97ee8a1.

dav3r reviewed (approved):
👍  aside from some capitalization foo that I mentioned.

mcdonnnj committed:
Fix unnecessary capitalization and refresh README

The term "security group" is not a proper noun and so it does not need to be
capitalized.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
